cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -72,6 +72,60 @@ void CallProfiler::print_profiling_result(const RadixTreeNode<CallerProfile> & n
     }
 }
 
+void CallProfiler::serialize(std::ostream & outstream) const
+{
+    outstream << R"({)";
+    outstream << R"("radix_tree": )";
+    serialize_radix_tree(outstream);
+    outstream << R"(})";
+}
+
+void CallProfiler::serialize_radix_tree(std::ostream & outstream) const
+{
+    outstream << R"({)";
+    outstream << R"("nodes": [)";
+    serialize_radix_tree_nodes(*(m_radix_tree.get_current_node()), true, outstream);
+    outstream << R"(],)";
+    outstream << R"("current_node": )" << m_radix_tree.get_current_node()->key() << R"(,)";
+    outstream << R"("unique_id": )" << m_radix_tree.get_unique_node();
+    outstream << R"(})";
+}
+
+// NOLINTNEXTLINE(misc-no-recursion)
+void CallProfiler::serialize_radix_tree_nodes(const RadixTreeNode<CallerProfile> & node, bool is_first_node, std::ostream & outstream) const
+{
+    // Avoid the trailing comma.
+    if (!is_first_node)
+    {
+        outstream << R"(,)";
+    }
+    outstream << R"({)";
+    outstream << R"("key": )" << node.key() << R"(,)";
+    outstream << R"("name": ")" << node.name() << R"(",)";
+    outstream << R"("data": )";
+    node.data().serialize(outstream);
+    outstream << R"(,)";
+    outstream << R"("children": [)";
+    bool is_first_child = true;
+    for (const auto & child : node.children())
+    {
+        // Avoid the trailing comma.
+        if (!is_first_child)
+        {
+            outstream << R"(,)";
+        }
+        is_first_child = false;
+        outstream << child->key();
+    }
+    outstream << R"(])";
+    outstream << R"(})";
+    for (const auto & child : node.children())
+    {
+        // NOLINTNEXTLINE(misc-no-recursion)
+        serialize_radix_tree_nodes(*child, false, outstream);
+    }
+}
```

Comment:
In order to avoid the multiprogramming, I implement these three functions under the CallProfiler.